staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller_debug.go

```diff
@@ -0,0 +1,199 @@
+}
+
+func (cfgCtl *configController) dumpPerPriorityLevel(w http.ResponseWriter, r *http.Request) {
+	cfgCtl.lock.Lock()
+	defer cfgCtl.lock.Unlock()
+	tabWriter := tabwriter.NewWriter(w, 8, 0, 1, ' ', 0)
+	columnHeaders := []string{
+		"PriorityLevelName",
+		"FlowSchemaName",
+		"ActiveQueues",
+		"IsIdle",
+		"RobinIndex",
+		"WaitingRequests",
+		"ExecutingRequests",
+	}
+	tabPrint(tabWriter, rowForHeaders(columnHeaders))
+	for _, plState := range cfgCtl.priorityLevelStates {
+		if plState.queues == nil {
+			continue
+		}
+		queueSetDigest := plState.queues.Dump()
+		activeQueueNum := 0
+		for _, q := range queueSetDigest.Queues {
+			if len(q.Requests) > 0 {
+				activeQueueNum++
+			}
+		}
+
+		tabPrint(tabWriter, rowForPriorityLevel(
+			plState.pl.Name,
+			activeQueueNum,
+			plState.queues.IsIdle(),
+			queueSetDigest.RobinIndex,
+			queueSetDigest.Waiting,
+			queueSetDigest.Executing,
+		))
+	}
+	runtime.HandleError(tabWriter.Flush())
+}
+
+func (cfgCtl *configController) dumpPerQueue(w http.ResponseWriter, r *http.Request) {
+	cfgCtl.lock.Lock()
+	defer cfgCtl.lock.Unlock()
+	tabWriter := tabwriter.NewWriter(w, 8, 0, 1, ' ', 0)
+	columnHeaders := []string{
+		"PriorityLevelName",
+		"PendingRequests",
+		"ExecutingRequests",
+		"RobinIndex",
+		"Index",
+		"Length",
+		"VirtualStart",
+	}
+	tabPrint(tabWriter, rowForHeaders(columnHeaders))
+	for _, plState := range cfgCtl.priorityLevelStates {
+		if plState.queues == nil {
+			continue
+		}
+		queueSetDigest := plState.queues.Dump()
+		if plState.queues == nil {
+			continue
+		}
+		for _, q := range queueSetDigest.Queues {
+			tabPrint(tabWriter, rowForQueue(
+				plState.pl.Name,
+				queueSetDigest.Waiting,
+				queueSetDigest.Executing,
+				queueSetDigest.RobinIndex,
+				q.Index,
+				len(q.Requests),
+				q.VirtualStart,
+			))
+		}
+	}
+	runtime.HandleError(tabWriter.Flush())
+}
+
+func (cfgCtl *configController) dumpPerRequest(w http.ResponseWriter, r *http.Request) {
+	cfgCtl.lock.Lock()
+	defer cfgCtl.lock.Unlock()
+	tabWriter := tabwriter.NewWriter(w, 8, 0, 1, ' ', 0)
+	columnHeaders := []string{
+		"PriorityLevelName",
+		"FlowSchemaName",
+		"QueueIndex",
+		"FlowDistingsherType",
+		"FlowDistingsher",
+		"ArriveTime",
+		"StartTime",
+	}
+	tabPrint(tabWriter, rowForHeaders(columnHeaders))
+	for _, plState := range cfgCtl.priorityLevelStates {
+		if plState.queues == nil {
+			continue
+		}
+		queueSetDigest := plState.queues.Dump()
+		if plState.queues == nil {
+			continue
+		}
+		for _, q := range queueSetDigest.Queues {
+			for _, r := range q.Requests {
+				tabPrint(tabWriter, rowForRequest(
+					plState.pl.Name,
+					r.MatchedFlowSchema,
+					q.Index,
+					r.FlowSchemaDistinguisherType,
+					r.FlowDistinguisher,
+					r.ArriveTime,
+					r.StartTime,
+				))
+			}
+		}
+	}
+	runtime.HandleError(tabWriter.Flush())
+}
+
+func tabPrint(w io.Writer, row string) {
+	_, err := fmt.Fprintf(w, row)
```

Comment:
This is taking a data string and treating it as a format, which is dangerous and unnecessarily expensive.

Reply:
updated to non-formatting print call